### DISCUSSION TITLE
Eddy identification on multiple files

### DISCUSSION THREAD

**Discussion by Mrvy29** [AntSimi/py-eddy-tracker#178] [Q&A] [answered]
Hi there:)

Recently found your product on Aviso + any differences from the identification here?
I have multiple files (for each day ) to run the eddy identification on and cannot seem to understand how to properly fit the date parameter. when running individual files this is date = datetime( 2016, 11, 02) for example.
I am trying a loop such as (parameter in question between "**"
for f in files:
g.eddy_identification("adt", "u", "v", *****date*****, 0.002, shape_error=55)
in my loop the list of identifications has the same date... 
I tried to find a solution on my own but maybe you can share your solution in the list of examples provided?

That would be extremely helpful !
Thank you !

**Comment by Mrvy29** [comment]
Hi , I have also asked the question on stack overflow: [Question](https://stackoverflow.com/questions/74545439/iterating-through-file-list-and-using-part-of-file-name-as-a-date-in-a-function?noredirect=1#comment131669931_74545439)

**Comment by AntSimi** [comment]
In the identification loop did you take a look at 
```python
print(a.time)
# Where a is anticyclonic given by eddy_identification
```

**Comment by Mrvy29** [comment]
Yes of course the problem is all identification has the same date coming
from the last file of the loop.. can you maybe share a pipeline for
identification on multiple files?
I tried a multidate .nc file but the function doesn’t allow for a list only
a single date time object.
Thanks so much for the support.

On Mon, 28 Nov 2022 at 18:51 Antoine Delepoulle ***@***.***>
wrote:

> In the identification loop did you take a look at
>
> print(a.time)# Where a is anticyclonic given by eddy_identification
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AntSimi/py-eddy-tracker/discussions/178#discussioncomment-4255393>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2ZPD6UBKW6MVJO22ELEU6TWKTPKXANCNFSM6AAAAAASHTN3DQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

[image: Seafloor Scene Icon]

**Comment by AntSimi** [comment, marked as answer]
Could you share the full script that you use?

**Reply by Mrvy29** [reply]
Sure, I would need to see how someone else has input multiple dates into eddy.identification() function to be helpful I guess.
```python
from datetime import datetime
from matplotlib import pyplot as plt
from numpy import arange
from py_eddy_tracker import data
from py_eddy_tracker.dataset.grid import RegularGridDataset
import glob as gl
import os

files = gl.glob('*.nc')
files
```

```
['dt_europe_allsat_phy_l4_20000117_20210726.nc',
 'dt_europe_allsat_phy_l4_20000125_20210726.nc',
 'dt_europe_allsat_phy_l4_20000130_20210726.nc',
 'dt_europe_allsat_phy_l4_20000102_20210726.nc',
 'dt_europe_allsat_phy_l4_20000107_20210726.nc',
 'dt_europe_allsat_phy_l4_20000129_20210726.nc',
 'dt_europe_allsat_phy_l4_20000112_20210726.nc',
 'dt_europe_allsat_phy_l4_20000120_20210726.nc']
```
```python
for f in files:
    date = datetime(int(os.path.splitext(f)[0][24:-13]),int(os.path.splitext(f)[0][28:-11]),
                    int(os.path.splitext(f)[0][30:-9]))
    print(date)
```
``` 
2000-01-17 00:00:00
2000-01-25 00:00:00
2000-01-30 00:00:00
2000-01-02 00:00:00
2000-01-07 00:00:00
2000-01-29 00:00:00
2000-01-12 00:00:00
2000-01-20 00:00:00
```

```python
for f in files:
    g = RegularGridDataset(f,"longitude","latitude")
    g.add_uv("adt")
    g.bessel_high_filter("adt", 500)
    date = datetime(int(os.path.splitext(f)[0][24:-13]),int(os.path.splitext(f)[0][28:-11]),
                    int(os.path.splitext(f)[0][30:-9]))
a, c = g.eddy_identification("adt", "u", "v", date , 0.002, shape_error=55)

print(a.time)
```

```
[18283. 18283. 18283. 18283. 18283. 18283. 18283. 18283. 18283. 18283.
 18283. 18283. 18283. 18283. 18283. 18283. 18283. 18283. 18283. 18283.
 18283. 18283. 18283. 18283. 18283. 18283. 18283.]
 ```
a.nc

  | amplitude | effective_area | effective_contour_height | effective_contour_latitude | effective_contour_longitude | effective_contour_shape_error | effective_radius | inner_contour_height | latitude | latitude_max | ... | num_point_s | speed_area | speed_average | speed_contour_height | speed_contour_latitude | speed_contour_longitude | speed_contour_shape_error | speed_radius | time | uavg_profile
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

0.0126 | 1.890812e+09 | -0.024 | 34.119999 | 21.690002 | 20.0 | 24950.0 | -0.012 | 34.310383 | 34.314022 | ... | 17 | 1.890812e+09 | 0.1034 | -0.024 | 34.119999 | 21.690002 | 20.0 | 24950.0 | 2000-01-22 | 0.1034
0.0126 | 1.890812e+09 | -0.024 | 34.110001 | 21.720001 | 20.0 | 24950.0 | -0.012 | 34.310383 | 34.314022 | ... | 17 | 1.890812e+09 | 0.1034 | -0.024 | 34.110001 | 21.720001 | 20.0 | 24950.0 | 2000-01-22 | 0.1033
0.0126 | 1.890812e+09 | -0.024 | 34.099998 | 21.760010 | 20.0 | 24950.0 | -0.012 | 34.310383 | 34.314022 | ... | 17 | 1.890812e+09 | 0.1034 | -0.024 | 34.099998 | 21.760010 | 20.0 | 24950.0 | 2000-01-22 | 0.1031
0.0126 | 1.890812e+09 | -0.024 | 34.099998 | 21.770004 | 20.0 | 24950.0 | -0.012 | 34.310383 | 34.314022 | ... | 17 | 1.890812e+09 | 0.1034 | -0.024 | 34.099998 | 21.770004 | 20.0 | 24950.0 | 2000-01-22 | 0.1029
0.0126 | 1.890812e+09 | -0.024 | 34.090000 | 21.790009 | 20.0 | 24950.0 | -0.012 | 34.310383 | 34.314022 | ... | 17 | 1.890812e+09 | 0.1034 | -0.024 | 34.090000 | 21.790009 | 20.0 | 24950.0 | 2000-01-22 | 0.1027`

**Reply by AntSimi** [reply]
You need to do identification for each time step(in the loop), and each `a` object contains identification for one grid, so all observation in `a` will have the same date

**Reply by Mrvy29** [reply]
Thanks I understand, would you be able to share an example of what I did wrong and how to do it right?

**Reply by AntSimi** [reply]
Your script is ok, just this two lines must be in your loop:
```python
a, c = g.eddy_identification("adt", "u", "v", date , 0.002, shape_error=55)
print(a.time)
```

**Reply by Mrvy29** [reply]
Oh but of course they are, it just didn't ident properly while pasting the code.

**Reply by AntSimi** [reply]
But in your display only one ouput are displaying.
Could you return output of this for each step of your loop:
```python
print(f, a.time)
```

**Reply by Mrvy29** [reply]
OK it seems to have solved it 👍  thanks a lot , I appreciate it
```
for f in files:
    g = RegularGridDataset(f,"longitude","latitude")
    g.add_uv("adt")
    g.bessel_high_filter("adt", 500)
    date = datetime(int(os.path.splitext(f)[0][24:-13]),int(os.path.splitext(f)[0][28:-11]),
                    int(os.path.splitext(f)[0][30:-9]))
    a, c = g.eddy_identification("adt", "u", "v", date , 0.002, shape_error=55)
    a_name = 'a'+date.strftime('%Y_%m_%d')+'.nc'
    c_name = 'c'+date.strftime('%Y_%m_%d')+'.nc'
    a.write_file(filename = a_name)
    c.write_file(filename = c_name)
```
